systems/data-plan/base-rate/pkg/db/rate_repo.go

```diff
@@ -0,0 +1,50 @@
+package db
+
+import (
+	"github.com/ukama/ukama/systems/common/sql"
+)
+
+type BaseRateRepo interface {
+	GetBaseRate(Id uint64) (*Rate, error)
+	GetBaseRates(country, network, effectiveAt, simType string) ([]Rate, error)
+	UploadBaseRates(rateList []Rate) error
+}
+
+type baseRateRepo struct {
+	Db sql.Db
+}
+
+func NewBaseRateRepo(db sql.Db) *baseRateRepo {
+	return &baseRateRepo{
+		Db: db,
+	}
+}
+
+func (u *baseRateRepo) GetBaseRate(rateId uint64) (*Rate, error) {
+	var rate *Rate
+	result := u.Db.GetGormDb().First(&rate, "Id=?", rateId)
```

Comment:
Mixing pointers and pointer on pointers.

Reply:
Addressed